### PR TITLE
Add module frame

### PR DESCRIPTION
**Pull request by tobes**
The frame module.  This allow modules to be grouped.  This is useful if you wish to add more than one module to a 'slot' in a group.

```
group {
    frame {
        volume_status {}
        battery_level {}
    }

    time {}
}
```

### CODE REVIEW

**Comment by ultrabug** [comment]
We ought to document this more than in the module itself since it's a great feature ! Would you care updating the documentation here https://github.com/ultrabug/py3status/tree/master/doc#group_modules with this feature please ?

**Comment by tobes** [comment]
Sure

**Comment by tobes** [comment]
I've added some documentation as requested.

I also uncovered a bug, that I fixed in this PR.  Some modules need to have the `output_modules` dict in core.py available to get content from other modules, but were running before it was built.  We now build the dict before starting the modules.

**Comment by tobes** [comment]
I'm thinking of adding an open/close button to frame like the group module has as I think this is a useful space saving feature.

Would that be acceptable?  Alternatively that could be a separate module but it feels that pushing it into frame would be more sensible.   Then any module you wanted to be open/close-able you could just put in a frame.

**Comment by ultrabug** [comment]
> I'm thinking of adding an open/close button to frame like the group module has as I think this is a useful space saving feature.

Sure

**Comment by tobes** [comment]
Ok, this now has the button added and a way to change the separators between the modules.

Anyhow I think I'm finished on this now.

I also added `py3.is_my_event(event)` helper that is useful for containers like group, hope that is ok.

File: py3status/modules/frame.py [hunk 1]
```diff
@@ -0,0 +1,141 @@
+# -*- coding: utf-8 -*-
+"""
+Group modules and treat as a single one.
```
Comment (ultrabug) [inline py3status/modules/frame.py]: "and display them as a single one" instead ?

**Comment by ultrabug** [comment]
minor docstring remark and I merge :+1: thanks a lot !

**Comment by tobes** [comment]
typo fixed

**Comment by ultrabug** [comment]
awesome :D

**Pull request merged**